docs/source/deep-learning/tensorflow/guide/index.rst

```diff
@@ -0,0 +1,269 @@
+for example if you don't want to log the dataset information, then you can run
+``mlflow.tensorflow.autolog(log_dataset_info=False)``. Please refer to the API documentation
+:py:func:`mlflow.tensorflow.autolog()` for full customization options.
+
+
+Understanding Autologging
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The way we autolog Tensorflow is by registering a custom callback to the Keras model via monkey patch.
+Briefly we attach a MLflow callback to the Keras model that works similarly to normal Keras callbacks.
+The callback will be triggered per ``every_n_iter`` epochs to log the training metrics, and log model
+information before training, also log the saved model after training completes.
+
+
+Logging to MLflow with Keras Callback
+--------------------------------------
+
+As discussed in the previous section, under the hood MLflow autologging is simply using a Keras
+callback. You can write your custom callback to select information to log to MLflow.
+
+Using the Predefined Callback
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+MLflow offers a predefined callback :py:class:`mlflow.tensorflow.MLflowCallback` that you can use or
+extend to log information to MLflow. The callback function provides the same functionality as autologging,
+and is suitable for users willing to have a better control of the experiment. Using ``mlflow.tensorflow.MLflowCallback``
+is the same as other Keras callbacks:
+
+.. code-block:: python
+
+    with mlflow.start_run():
+        model.fit(
+            data,
+            label,
+            batch_size=5,
+            epochs=2,
+            callbacks=[mlflow.tensorflow.MLflowCallback()],
+        )
+
+You can change the logging frequency in :py:class:`mlflow.tensorflow.MLflowCallback` by setting
+``log_every_epoch`` and ``log_every_n_steps``, by default metrics are logged per epoch. Please refer to
+the API documentation for more details.
+
+Customizing MLflow Logging
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also write your own callback to log information to MLflow. To do that, you need to define
+a class subclassing from `keras.callbacks.Callback <https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback>`_,
+which provides hooks at various stages of training and validation, e.g., ``on_epoch_end`` and
+``on_train_end`` are called separately at the end of each epoch and when the training is finished.
+You can then use the callback in ``model.fit()``. Here is a simple example for logging the training metrics
+in log scale:
+
+.. code-block::
+
+    from tensorflow import keras
+    import math
+    import mlflow
+
+    class MLflowCallback(keras.callbacks.Callback):
+        def on_epoch_end(self, epoch, logs=None):
+            logs = logs or {}
+            for k, v in logs.items():
+                mlflow.log_metric(f"log_{k}", math.log(v), step=epoch)
+
+Briefly ``logs`` object will contain your training metrics (``loss`` and ``metrics`` as defined in ``model.compile()``),
```

Comment:
Maybe Briefly can be changed to After each epoch for better clarity? and also I think the training and validation metrics can be reworded to be clearer?

Reply:
sg, done!